SplashScreen/app/src/main/java/com/example/android/splashscreen/MainActivity.kt

```diff
@@ -0,0 +1,96 @@
+package com.example.android.splashscreen
+
+import android.app.UiModeManager
+import android.os.Bundle
+import android.os.Process
+import android.view.View
+import android.view.ViewTreeObserver
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.updatePadding
+import com.example.android.splashscreen.databinding.MainActivityBinding
+
+/**
+ * Shows the app content that is commonly used in all of [DefaultActivity], [AnimatedActivity], and
+ * [CustomActivity]. This also handles the custom dark mode.
+ */
+abstract class MainActivity : AppCompatActivity() {
+
+    private val viewModel: MainViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val binding = MainActivityBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setSupportActionBar(binding.toolbar)
+        // Suspend drawing the app content until the initial data is ready.
+        suspendDraw()
```

Comment:
I'd call this `suppressDraw()` instead. `suspend` makes me think of coroutines.

Reply:
Renamed.